### PR TITLE
FUSETOOLS-1955 - Fix Go into feature

### PR DESCRIPTION
**Pull request by apupier**
- avoid NPE
- ensure that the selectedContainer attribute is updated when "going up"

I don't see how to write meaningful test except by writing a UI one.

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelDesignEditor.java [hunk 1]
```diff
@@ -449,6 +449,9 @@ public AbstractCamelModelElement getSelectedContainer() {
 	public void setSelectedContainer(AbstractCamelModelElement route) {
 		this.selectedContainer = route;
 		switchContainer();
+		if(selectedContainer == null){
```
Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelDesignEditor.java]: Great catch here.

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java [hunk 2]
```diff
@@ -75,16 +75,15 @@ private boolean isGoUp(AbstractCamelModelElement cme, CamelDesignEditor editor)
...
+		return cme instanceof CamelRouteElement && cme.equals(editor.getSelectedContainer());
 	}
 	
 	private boolean isGoInto(AbstractCamelModelElement cme, CamelDesignEditor editor) {
 		// go into is allowed if:
 		// - selected element is a route
 		// - selected container is the camel context
 		// - context contains more than one route
-		return 	cme.getNodeTypeId().equalsIgnoreCase("route") && 
+		return 	"route".equalsIgnoreCase(cme.getNodeTypeId()) &&
```
Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java]: Do we want to externalize this string or mark it as not externalizable? (i.e.  //$NON-NLS-1$ )

**Comment by bfitzpat** [comment]
+1 except with the externalized string question

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java (hunk 2 above)
Comment (apupier) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java]: even better, reusing Constants :-)

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java [hunk 3]
```diff
@@ -83,7 +83,7 @@ private boolean isGoInto(AbstractCamelModelElement cme, CamelDesignEditor editor
 		// - selected element is a route
 		// - selected container is the camel context
 		// - context contains more than one route
-		return 	"route".equalsIgnoreCase(cme.getNodeTypeId()) && 
+		return 	AbstractCamelModelElement.ROUTE_NODE_NAME.equalsIgnoreCase(cme.getNodeTypeId()) &&
```
Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/GoIntoContainerFeature.java]: Looks great! Thanks Aurelien

**Comment by bfitzpat** [comment]
+1 with the constant fix. :)

**Pull request merged**